### PR TITLE
workflow: fix cache

### PR DESCRIPTION
**Pull request by FeelyChau**
1. We have refactored package manager to `yarn` from `npm`, so the lock file name changed from `package-lock.json` to `yarn.lock`, and only the root directory has the lock file.
2. The right way to use function `hashFiles` is: `hashFiles('fileA.json', 'fileB.json')`.

### CODE REVIEW

**Comment by yorkie** [comment]
@FeelyChau May I ask you to write a bit about what to fix and how in this PR?

**Comment by FeelyChau** [comment]
> @FeelyChau May I ask you to write a bit about what to fix and how in this PR?

OK, updated.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -45,10 +45,10 @@ jobs:
           */*/node_modules
           packages/boa/pybind11
           packages/boa/pyodide
-        key: ${{ runner.os }}-${{ matrix.node_version }}-${{ hashFiles('**/package-lock.json,packages/boa/pybind11/Makefile,packages/boa/pyodide/Makefile') }}
+        key: ${{ runner.os }}-${{ matrix.node_version }}-yarn-${{ hashFiles('**/yarn.lock', 'packages/boa/pybind11/Makefile', 'packages/boa/pyodide/Makefile') }}
```
Comment (yorkie) [inline .github/workflows/build.yml:48]: Why update the key with appended yarn?

Comment (FeelyChau) [reply .github/workflows/build.yml:48]: YARN is a completely different package manager and the cache is not compatible.

**Review by WenheLI — APPROVED** [review]
LGTM

File: .github/workflows/build.yml (hunk 1 above)
Comment (yorkie) [reply .github/workflows/build.yml:48]: But the file is different so that the key has been updated, though.

**Review by yorkie — APPROVED** [review]
Looks good to me

**Pull request merged**